crates/core/tedge/src/cli/common.rs

```diff
@@ -1,50 +1,107 @@
-use anyhow::bail;
+use anyhow::Context;
 use std::borrow::Cow;
-use std::str::FromStr;
+use std::fmt;
 use tedge_config::system_services::SystemService;
 use tedge_config::ProfileName;
-use yansi::Paint;
 
-pub type Cloud = MaybeBorrowedCloud<'static>;
+#[derive(clap::Args, PartialEq, Eq, Debug, Clone)]
+pub struct CloudArgs {
```

Comment:
A comment would help to understand when to use `CloudArgs` vs `OptionalCloudArgs` vs `Option<ProfileName>` as a clap argument.

- `tedge cert create` uses `OptionalCloudArgs` letting the user creates a certificate for a specific cloud profile as well as a generic one not specifically attached to some cloud profile.
- `tedge connect` uses `CloudArgs` because a cloud is required, possibly with a specific profile.
- `tedge cert upload c8y` uses `Option<ProfileName>` because this feature is only available for Cumulocity.

Reply:
I've now fixed this by changing this to be a subcommand, so we can wrap the result is in `Option`. I guess the point around `cert upload` is still relevant, but I think it's fairly obvious why that is a special case.